src/main/java/core/MyWebDriverManager.java

```diff
@@ -0,0 +1,85 @@
+ * license agreement you entered into with Fundacion Jala
+ */
+
+package core;
+
+import io.github.bonigarcia.wdm.WebDriverManager;
+import io.github.bonigarcia.wdm.config.DriverManagerType;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.lang.reflect.InvocationTargetException;
+
+import static core.utils.LoadEnvironmentFile.getDotenv;
+
+public class MyWebDriverManager {
+    private final DriverManagerType driverManagerType;
+    private static MyWebDriverManager myWebDriverManager;
+    private static WebDriver driver;
+    private static WebDriverWait wait;
+
+    /**
+     * Gets the web driver manager or creates one when needed.
+     *
+     * @param driverManagerType an enum with the driver type
+     * @return an instance of myWebDriverManager
+     */
+    public static MyWebDriverManager getWebDriverManager(final DriverManagerType
+                                                                 driverManagerType)
+            throws ClassNotFoundException, NoSuchMethodException,
+            InvocationTargetException, InstantiationException, IllegalAccessException {
+        if (myWebDriverManager == null) {
+            myWebDriverManager = new MyWebDriverManager(driverManagerType);
+            WebDriverManager.getInstance(driverManagerType).setup();
+            Class<?> driverClass =  Class.forName(driverManagerType.browserClass());
+            driver = (WebDriver) driverClass.getDeclaredConstructor().newInstance();
+            wait = new WebDriverWait(driver, Long.parseLong(getDotenv().get("EXPLICIT_WAIT_TIME")));
+        }
+        return myWebDriverManager;
+    }
+
+    /**
+     * Gets the driver instance according to its type.
+     *
+     * @return the Web Driver
+     * @throws ClassNotFoundException when the class does not exist
+     * @throws NoSuchMethodException when the method is not found
+     * @throws InvocationTargetException when the target can not be invoked
+     * @throws InstantiationException when it is not possible to create an instance
+     * @throws IllegalAccessException when it can not be accessed
+     */
+    public WebDriver getDriver() throws ClassNotFoundException, NoSuchMethodException,
+            InvocationTargetException, InstantiationException, IllegalAccessException {
+        return driver;
+    }
+
+    /**
+     * Gets the web driver wait instance according to its type.
+     *
+     * @return the web driver wait
+     * @throws ClassNotFoundException when the class does not exist
+     * @throws InvocationTargetException when the target can not be invoked
+     * @throws NoSuchMethodException when the method is not found
+     * @throws InstantiationException when it is not possible to create an instance
+     * @throws IllegalAccessException when it can not be accessed
+     */
+    public WebDriverWait getWebDriverWait() throws ClassNotFoundException,
```

Comment:
The throws may not be necessary

Reply:
The method I am implementing requires them. But I will work to change it to a try and catch on a future PR.